source/non_matching/mesh_classifier.cc

```diff
@@ -0,0 +1,531 @@
+         */
+        const SmartPointer<const DoFHandler<dim>> dof_handler;
+
+        /**
+         * Pointer to the vector global dof values of the level set function.
+         */
+        const SmartPointer<const VECTOR> level_set;
+      };
+
+
+
+      template <int dim, class VECTOR>
+      DiscreteLevelSetDescription<dim, VECTOR>::DiscreteLevelSetDescription(
+        const DoFHandler<dim> &dof_handler,
+        const VECTOR &         level_set)
+        : dof_handler(&dof_handler)
+        , level_set(&level_set)
+      {}
+
+
+
+      template <int dim, class VECTOR>
+      void
+      DiscreteLevelSetDescription<dim, VECTOR>::get_local_level_set_values(
+        const typename Triangulation<dim>::active_cell_iterator &cell,
+        Vector<double> &local_levelset_values)
+      {
+        typename DoFHandler<dim>::active_cell_iterator cell_with_dofs(
+          &dof_handler->get_triangulation(),
+          cell->level(),
+          cell->index(),
+          dof_handler);
+
+        // Get the dofs indices associated with the face.
+        const unsigned int n_dofs_per_cell =
+          cell_with_dofs->get_fe().n_dofs_per_cell();
+        std::vector<types::global_dof_index> dof_indices(n_dofs_per_cell);
+        cell_with_dofs->get_dof_indices(dof_indices);
+
+        local_levelset_values.reinit(dof_indices.size());
+
+        for (unsigned int i = 0; i < dof_indices.size(); i++)
+          local_levelset_values[i] =
+            dealii::internal::ElementAccess<VECTOR>::get(*level_set,
+                                                         dof_indices[i]);
+      }
+
+
+
+      template <int dim, class VECTOR>
+      const hp::FECollection<dim> &
+      DiscreteLevelSetDescription<dim, VECTOR>::get_fe_collection() const
+      {
+        return dof_handler->get_fe_collection();
+      }
+
+
+
+      template <int dim, class VECTOR>
+      void
+      DiscreteLevelSetDescription<dim, VECTOR>::get_local_level_set_values(
+        const typename Triangulation<dim>::active_cell_iterator &cell,
+        const unsigned int                                       face_index,
+        Vector<double> &local_levelset_values)
+      {
+        typename DoFHandler<dim>::active_cell_iterator cell_with_dofs(
+          &dof_handler->get_triangulation(),
+          cell->level(),
+          cell->index(),
+          dof_handler);
+
+        // Get the dofs indices associated with the face.
+        const unsigned int n_dofs_per_face =
+          dof_handler->get_fe().n_dofs_per_face();
+        std::vector<types::global_dof_index> dof_indices(n_dofs_per_face);
+        cell_with_dofs->face(face_index)->get_dof_indices(dof_indices);
+
+        local_levelset_values.reinit(dof_indices.size());
+
+        for (unsigned int i = 0; i < dof_indices.size(); i++)
+          local_levelset_values[i] =
+            dealii::internal::ElementAccess<VECTOR>::get(*level_set,
+                                                         dof_indices[i]);
```

Comment:
I guess for this one you still need the present code, as I can't see that the other function would fit.